### DISCUSSION TITLE
use of undefined type 'nlohmann::detail::wide_string_input_helper<BaseInputAdapter,16>' compiler Error C2027

### DISCUSSION THREAD

**Discussion by michele-nuzzi** [nlohmann/json#2780] [Q&A] [answered]
``hi! 
I'm having some troubles in visual Studio 2019 using your code; as you can read from the title of the discussion I've got the following error while compiling:
`Error	C2027	use of undefined type 'nlohmann::detail::wide_string_input_helper<BaseInputAdapter,16>' ..\nlohmann\detail\input\input_adapters.hpp	315	
`
since the program doesn't even compile I have no idea of how to interpret the error;
if it can help I'm sure the json received is utf-8 encoded.

Could you please help me?

**Comment by nlohmann** [comment, marked as answer]
In order to asses your issue, we need the following information:

- What is the issue you have?

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

- What is the expected behavior?

- And what is the actual behavior instead?

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

- Did you use a released version of the library or the version from the `develop` branch?

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

**Reply by michele-nuzzi** [reply]
Hi!
I'm sorry if the question wasn't well-poned, I hope the next information I got can clear things out:

since I've just started the project, I could afford to try commenting-out code and see which part was causing the issue; it turned out the following simple function was the problem.
```
using Json = nlohmann::json;

Json removeOuterStringIfAny(const Json& possiblyWrappedJson)
{
	if (possiblyWrappedJson.is_string()) return Json::parse(possiblyWrappedJson);
	return possiblyWrappedJson;
}
```
as you can imagine the expected behaviour is to get a json object or array from a json whose internal type is a string;
the actual behaviour instead is in fact the error discussed previously.

a possible solution I went for is the following:
```
Json removeOuterStringIfAny(const Json& possiblyWrappedJson)
{
	if (possiblyWrappedJson.is_string()) return Json::parse(std::string(possiblyWrappedJson));
	return possiblyWrappedJson;
}
```
however, I'd like to know if there is some better way to achieve the same result.

thank you for your response anyway, following your step helped me a lot!
